flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -1498,11 +1505,35 @@ void AccAttributeVisitor::CheckMultipleAppearances(
     AddDataSharingAttributeObject(*target);
   }
 }
+std::optional<int64_t> OmpAttributeVisitor::GetMaskedTId(
```

Comment:
Does this work correctly if there is a variable or expression in the filter clause?

Reply:
I was wrong to treat it as a constant expression. Thanks for pointing this out. I have removed the changes.